krishnankit/shopping_cart/code/main.rb

```diff
@@ -0,0 +1,41 @@
+load "service.rb"
+products = []
+cart = []
+
+
+while 1
+  display_menu()
+  print "Your option: "
+  user_input = gets.chomp.to_i
+  puts "Mocked."
+  case user_input
+    when 0
+      break
+    when 1
+      display_products(products)
+    when 2
+      print "Enter the name: "
+      name = gets.chomp.downcase
+      print "Enter the price: "
+      price = gets.chomp.to_i
+      add_product(products, name, price)
+    when 3
+      print "Enter the name: "
+      name = gets.chomp.downcase
+      print "Enter the quantity: "
+      quantity = gets.chomp.to_i
+      add_to_cart(products, cart, name, quantity)

```

Comment:
when you enter the name of product which doesn't exist in the inventory then it should give an error message that the product is not found rather than asking for quantity because it's not make sense for asking quantity if that product is not in inventory

Reply:
Added the condition.